### ISSUE TITLE
[bug] |CPH truncates files.

### ISSUE THREAD

**Issue by egery** [realmml/X-Mass-ROM#5]
I've tried to copy an Orgams source ".o" with |CPH, but only first few bytes were copied instead of the whole 48k file.
This is an headerless binary file, so |CPH should not take EOF flag (triggered by &1A byte) into account but rely on physical size as given by UTOPIA's |INFO.

**Comment by realmml** [comment]
Yes, this "bug" is known, and only it will be fixed in a future/less hacky version of |CPH. For now, if a file has not a correct header checksum, then CPH thinks that it's an ASCII file without header and it will respect the EOF flag for not copying any extra trash byte.

**Comment by realmml** [comment]
I am not so sure, there is a lot of files available in a bad state (bad checksums and wrong filelengths) and i would prefer not supporting that emulator practices anymore... even i would love to put the really important Amsdos file header fields directly in the FAT entries and saving the boring task of handling the 2 headers (FAT entry and Amsdos).

This |CPH RSX should be easy to fix (for example, checking if the bytes read are equal to the file size in the file FAT entry), but i don't know when i will get more time for working in this, when there is a lot more important things as wildcards support.

**Comment by egery** [comment]
Thanks for you answer!

- I don't see how it is related to emulator. The physical size is the number of &80-long chunks as encoded in AMSDOS entry. It comes from CP/M and even if not byte-precise, it's the best indicator of real filesize, and the only one for binary "ASCII" (read: headerless) files such as ".o" ones. That dictates the size occupied on disc, and that's why it's the one that should be taken into account for copy.

- For me it's clearly a bug, since most copy tools (PARADOS, crime, ...) rightly handle this case. And I think a bug is always more critical than new features!

- By the way, for maximum compatibility, you should save the whole amsdos header anyway.
